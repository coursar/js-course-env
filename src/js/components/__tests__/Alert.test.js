/**
 * @jest-environment jsdom
 */
import createAlert from '../Alert';

test('should render defaults', () => {
  // (A)rrange
  const containerEl = document.createElement('div');

  // (A)ct
  const alertEl = createAlert();
  containerEl.appendChild(alertEl);

  // (A)ssert
  expect(containerEl).toMatchSnapshot();
});

test('should render args', () => {
  // (A)rrange
  const containerEl = document.createElement('div');

  // (A)ct
  const alertEl = createAlert({
    message: 'Network error',
    image: 'neutral.svg',
  });
  containerEl.appendChild(alertEl);

  // (A)ssert
  expect(containerEl).toMatchSnapshot();
});
