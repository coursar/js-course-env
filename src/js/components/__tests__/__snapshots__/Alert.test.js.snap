// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render args 1`] = `
<div>
  

  <div
    class="alert alert-danger"
  >
    
  
    <span
      data-id="message"
    >
      Network error
    </span>
    
  
    <img
      data-id="image"
      src="neutral.svg"
    />
    

  </div>
  

</div>
`;

exports[`should render defaults 1`] = `
<div>
  

  <div
    class="alert alert-danger"
  >
    
  
    <span
      data-id="message"
    />
    
  
    <img
      data-id="image"
      src="error.svg"
    />
    

  </div>
  

</div>
`;
